File: Question_2/Drone.py
from UAV import UAV
#subclass
class Drone(UAV):
    def __init__(self, name):
        self.name = name

    def takeoff(self):
        #if landing don't change anything
        if(self.is_landing):
            return False

        self.is_takeoff = False if self.is_landing else True
        return self.is_takeoff

File: Question_3/Drone.cpp
#include "UAV.h"
#include <iostream>
#include <string>

using namespace std;

class Drone : public UAV{
    //constructor
    public:
        Drone(string new_name){
            this->name = new_name;
        }

        bool takeoff(){
            if(this->is_landing){
                return false;
            }
            //ternary
            this->is_takeoff = this->is_takeoff ? false : true;
            return is_takeoff;
        }
};